qemu: support advanced hard drive configurations

a13xp0p0v opened the pull request:
Currently the 'image_device' config option in qemu.go doesn't properly
support the modern '-drive' argument for describing qemu hard disks.

In fact the various old ways to define qemu drives all boil down to
the common form '-drive if=TYPE,bus=BUS,unit=UNIT,OPTS...'

For example '-hda img' is equivalent to '-drive index=0,media=disk,file=img'.

Let's make the 'image_device' config option support both forms.

Signed-off-by: Alexander Popov <alex.popov@linux.com>

xairy commented:
Could you also document this somewhere (probably in the comments in `vm/qemu/qemu.go`) with examples for both variants.

a13xp0p0v commented:
Hello @xairy, 
Thanks for having a look!
I've added a verbose comment describing the 'image_device' config option in qemu.go.
I've also added a separate commit with style fixes that were automatically generated by `make presubmit`.
